### ISSUE TITLE
More robust handling of custom hosts

### ISSUE THREAD

**Issue by kevgo** [Originate/git-town#743]
As a Git Town user
I want that it understands my custom Git repository setup reliably
So that I can use it with every hosted Github server.
- read the ssh identities from the SSH keychain file to get the actual hostname that is associated with the custom host

refinement of #740

**Comment by charlierudolph** [comment]
Closing as no longer relevant after #1040
